samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -0,0 +1,205 @@
+          "{ct container   |mp4        | container name (supported: 'mp4', 'mov', 'avi', 'mkv') }"
+          "{cd codec       |h264       | codec name     (supported: 'h264', 'h265', 'mpeg4', 'mjpeg', 'vp8') }"
+          "{f file path    |           | path to file }";
+
+    CommandLineParser parser(argc, argv, keys);
+
+    parser.about("This program shows how to read a video file with GStreamer pipeline with OpenCV.");
+
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    if (!parser.check())
+    {
+        parser.printErrors();
+        return 0;
+    }
+
+    bool   arg_show;      // show frames on the screen
+    string arg_pipeline,  // GStreamer pipeline type
+           arg_container, // container type
+           arg_codec,     // codec type
+           arg_file_name, // path to videofile
+           codec_name,
+           meta_info;
+    VideoCapture cap;
+
+    arg_show      = parser.get<bool>  ("show");
+    arg_pipeline  = parser.get<string>("pipeline");
+    arg_container = parser.get<string>("container");
+    arg_codec     = parser.get<string>("codec");
+    arg_file_name = parser.get<string>("file");
+
+    // Check support plugins, containers and codecs
+    checkSupportArguments(parser);
+
+    // Choose the constructed GStreamer pipeline
+    if (arg_pipeline.find("gst") != string::npos)
+    {
+        ostringstream pipeline;
+        pipeline << "filesrc location=\"" << arg_file_name << "\"";
+        pipeline << " ! " << getGstDemuxPlugin(parser);
+
+        if (arg_pipeline.find("default") != string::npos)
+        {
+            pipeline << " ! decodebin";
+        }
+        else if (arg_pipeline.find("vaapi") != string::npos)
+        {
+            pipeline << " ! vaapidecodebin disable-vpp=true";
+            pipeline << " ! autovideoconvert";
+        }
+        else if (arg_pipeline.find("libav") != string::npos)
+        {
+            pipeline << " ! " << getGstAvDecodePlugin(parser);
+        }
+
+        pipeline << " ! videoconvert";
+        pipeline << "     dither=0";
+        pipeline << "     dither-quantization=0";
+        pipeline << "     chroma-resampler=0";
+        pipeline << "     alpha-mode=1";
+        pipeline << "     alpha-value=0";
+        pipeline << "     chroma-mode=3";
+        pipeline << " ! appsink sync=false";
+        cap = VideoCapture(pipeline.str(), CAP_GSTREAMER);
+    }
+    else if (arg_pipeline == "ffmpeg")
+    {
+        cap = VideoCapture(arg_file_name, CAP_FFMPEG);
+    }
+    else
+    {
+        parser.printMessage();
+        cout << "Not find GStreamer pipeline or FFMpeg" << endl;
+        return -4;
+    }
+
+    // Choose a show video or only mesuare fps
+    cout << "_____________________________________" << '\n';
+    cout << "Start measure frame per seconds (fps)" << '\n';
+    cout << "Loading ..." << '\n';
+
+    Mat frame;
+    unsigned long count_frame = 0;
```

Comment:
This variable is not needed because you can get frame number by calling `tick_counts.size()`.